Release notes v2.2.0 (#6242)
grafana/k6: release notes/v2.2.0.md

k6 `v2.2.0` is here 🎉! This release includes:

- `k6 cloud run --local-execution` now streams k6's logs to Grafana Cloud, so the test run's log view works for local execution too.
- `chromium.connectOverCDP()`, which connects browser tests to an already-running Chromium instance.
- `TextEncoder` and `TextDecoder` available as globals, and `WritableStream` support in `k6/experimental/streams`.
- A `k6 cloud load-zone list` command.
- Two new experimental feature flags: `merge-run-tags` and `freeze-env`.

## Breaking changes

There are no breaking changes in this release.

## New features

### `k6 cloud run --local-execution` streams logs to Grafana Cloud [#6171](https://github.com/grafana/k6/pull/6171)

> When a cloud test runs locally with `k6 cloud run --local-execution`, k6's logs now stream to the Grafana Cloud test run, so the run's log view is populated the same way it is for cloud execution. Previously, local-execution logs stayed on the machine running k6 and never reached the cloud. Secrets are redacted before logs are pushed, and the new `--no-cloud-logs` flag opts out:

Author: ```suggestion
When a cloud test runs locally with `k6 cloud run --local-execution`, k6's logs now stream to the Grafana Cloud test run, so the run's log view is populated the same way it is for cloud execution. Previously, local-execution logs stayed on the machine running k6 and never reached the cloud. Work with Grafana's secrets management to safely work with secrets and redact them if they're accidentally leaked into logs are pushed. Use the new `--no-cloud-logs` flag opts out to opt out of streaming of logs when working with `--local-execution`:
```

```shell
k6 cloud run --local-execution script.js
k6 cloud run --local-execution --no-cloud-logs script.js
```

### Connect to a running browser with `chromium.connectOverCDP()` [#6165](https://github.com/grafana/k6/pull/6165)

> The browser module can now attach to an existing Chromium-based browser over the Chrome DevTools Protocol, mirroring Playwright's `browserType.connectOverCDP()`. Pass the browser's WebSocket endpoint and k6 manages the returned browser's lifecycle — it's auto-closed at the end of the iteration, though you can call `close()` earlier to release the connection on demand.

Author: ```suggestion
The browser module can now attach to an existing Chromium-based browser over the Chrome DevTools Protocol, mirroring Playwright's `browserType.connectOverCDP()`. Pass the browser's WebSocket endpoint and k6 manages the returned browser's connection — it's auto-closed at the end of the iteration, though you can call `close()` earlier to release the connection on demand.
```

k6 `v2.2.0` is here 🎉! This release includes:

> - A revamped `k6 cloud run --local-execution` flow — runs are provisioned through the new Grafana Cloud k6 provisioning API, logs stream to Grafana Cloud, and externally-provisioned runs are supported.

Author: ```suggestion
- A revamped `k6 cloud run --local-execution` flow — runs are provisioned through the new Grafana Cloud k6 provisioning API, and logs stream to Grafana Cloud.
```

- `chromium.connectOverCDP()`, which connects browser tests to an already-running Chromium instance.
- `TextEncoder` and `TextDecoder` available as globals, and `WritableStream` support in `k6/experimental/streams`.

> - A configurable `handleSummary()` timeout and a `k6 cloud load-zone list` command.

Author: ```suggestion
- A `k6 cloud load-zone list` command.
```
